### ISSUE TITLE
Bring back job control

### ISSUE THREAD

**Issue by xiaq** [elves/elvish#410]

**Comment by krader1961** [comment]
I am a thumbs down on this proposal. I started programming in 1977 and eagerly embraced UNIX in the 1980's when interacting with a computer running UNIX was almost always over a serial connection and you rarely had more than a single connection (i.e., terminal session) to the system. Shell job control was essential in that world. Especially if you were interacting with a system via dial-up connection. Like I did as a Sequent Computer Systems support technician when interacting with its customer's systems in the 1980's and 1990's.

In 2020 the need for POSIX style job control is limited to situations like those mentioned above. And those situations are exceedingly rare today. It is exceedingly rare for an Elvish user to actually need POSIX shell job control and that will only decrease in the future. Also, given the quirks of multi-threaded programs (whether written in Go or C using POSIX threads) on UNIX like systems it is impossible to do this correctly 100% of the time.

**Comment by zzamboni** [comment]
I agree with @krader1961 on this. Normally if I need a new session, I just open a new tab in my terminal. Even remotely (which, admittedly, I don't do much these days), I'd rather use `screen` or `tmux` to handle multiple sessions, rather than rely on job control.

While I think it would be cool to have, I think the effort to get it right would not be worth it. Like *cough* https://github.com/elves/elvish/issues/1039 *cough* ;)

**Comment by andrewbaxter** [comment]
I use job control a lot for watching logs while running commands, I'd like this feature.

**Comment by IcedQuinn** [comment]
I mostly use job control for

1. A GUI process i nontheless have to start in a shell and then disconnect and throw away, or,
2. Pausing a text editor (vim, kak) to run a command with the intent to return to the editor briefly after

I use more tabs/terminal windows for multiple ongoing things. But the quick flip between editor and running a command is what crashes me in to the wall when i looked at elvish.
